### PR TITLE
chore: remove unused dependencies

### LINKED ISSUE #1598: Check dependencies of all crates
### Description

Over the years there were several refactors, which the lead to dependencies in Cargo.toml files which aren't actually used. I propose looking into all crates and check which dependencies are still needed. Some might also be only needed as dev-dependency (similarly to what I did at https://github.com/filecoin-project/rust-fil-proofs/pull/1593).

### Acceptance criteria

All crates specify only the dependencies in their Cargo.toml they actually need.

### Risks + pitfalls

I don't see risks, I think our test coverage on CI is good enough.

### Where to begin

There is tooling to check for unused dependencies. From my experience they don't always work well. This is what I did for the PR mentioned above:

 - Comment out all dependencies and run `cargo build`. Comment the normal dependencies in again until the build works again.
 - Run `cargo test` and start to comment in the dev-dependencies until the tests compile.
 - Check CI that everything still works.

### PR DESCRIPTION
**Pull request by 0x5459**
Remove unused dependencies. 
resolves #1598 .

### CODE REVIEW

**Review by dignifiedquire — APPROVED** [review]

File: fr32/Cargo.toml [hunk 1]
```diff
@@ -26,8 +25,8 @@ rand_xorshift = "0.3"
 
 [features]
 default = []
-cuda = ["bellperson/cuda"]
-opencl = ["bellperson/opencl"]
+cuda = []
+opencl = []
```
Comment (vmx) [inline fr32/Cargo.toml:29]: You can remove the whole features section.

**Review by vmx — COMMENTED** [review]
Just a minor thing, to clean things up even more.

File: fr32/Cargo.toml (hunk 1 above)
Comment (0x5459) [reply fr32/Cargo.toml:29]: ok.

**Review by vmx — APPROVED** [review]
Thanks!

**Pull request merged**